### ISSUE TITLE
Optimize text_classifer model memory usage

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#21960]
Related issue: https://github.com/brave/brave-browser/issues/21602
Parsed text classifier model (component id : `emgmepnebbddgnkhfmhdhmjifkglkamo`) takes **~24mb** of memory in Ads Service process.

It's looks excessive and we should find a way to make it smaller.
Another problem is that peak usage on start is more than **200mb.**

Here is the code that produce 24mb memory allocation: https://github.com/brave/brave-core/blob/master/vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/text_processing.cc#L63

**Comment by atuchin-m** [comment]
For QA:
1. check that ads text classification works well.
2. check memory consumption of ads service browser (for non-ios) or main process (for ios) is reduced.

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.39.87 Chromium: 101.0.4951.41 (Official Build) beta (64-bit)
-- | --
Revision | 93c720db8323b3ec10d056025ab95c23a31997c9-refs/branch-heads/4951@{#904}
OS | Ubuntu 18.04 LTS

### Reproduced the issue in 1.38.x

Verify that the issuers were downloaded
<details>

```
[18839:18839:0509/225506.081261:VERBOSE1:issuers.cc(80)] OnGetIssuers
[18839:18839:0509/225506.081836:VERBOSE6:issuers.cc(82)] URL Response:
  URL: https://ads-serve.brave.com/v1/issuers/
  Response Status Code: 200
```

</details>

Verify that unblinded tokens are downloaded
<details>

```

[18839:18839:0509/225526.441519:VERBOSE1:refill_unblinded_tokens.cc(323)] Added 50 unblinded tokens, you now have 50 unblinded tokens
[18839:18839:0509/225526.441569:VERBOSE1:account.cc(292)] Successfully refilled unblinded tokens


```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (US) - Version: 1.0.803
Brave Ads Resources (US) - Version: 1.0.47
Brave Ads Resources (en) - Version: 1.0.41

![image](https://user-images.githubusercontent.com/34715963/167600282-e85c5cf3-2faf-4bfd-8741-5785d44519e2.png)

</details>

Viewed an ad and dimissed it

#### The memory used is 37352KB

![image](https://user-images.githubusercontent.com/34715963/167609414-d35c64e0-9be6-4225-8677-1ff5eeec4c39.png)


### Verified fixed in 1.39.x

Verify that the issuers were downloaded
<details>

```

[19550:19550:0510/001142.312340:VERBOSE1:issuers.cc(80)] OnGetIssuers
[19550:19550:0510/001142.312388:VERBOSE6:issuers.cc(82)] URL Response:
  URL: https://static.ads.brave.com/v1/issuers/
  Response Status Code: 200


```


</details>

Verify that unblinded tokens are downloaded
<details>

```

[19550:19550:0510/001153.082434:VERBOSE1:refill_unblinded_tokens.cc(322)] Added 50 unblinded tokens, you now have 50 unblinded tokens
[19550:19550:0510/001153.082709:VERBOSE1:account.cc(291)] Successfully refilled unblinded tokens


```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (US) - Version: 1.0.803
Brave Ads Resources (US) - Version: 1.0.47
Brave Ads Resources (en) - Version: 1.0.41

![image](https://user-images.githubusercontent.com/34715963/167618277-64cdce6f-135b-4a47-9827-a417742541fc.png)

</details>

Viewed an ad and dimissed it

#### The memory used is 19252KB, which is less than in 1.38.x


![image](https://user-images.githubusercontent.com/34715963/167618400-63c61ba1-e8ec-45c0-9ae1-feec852693de.png)

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.39.96 Chromium: 101.0.4951.54 (Official Build) beta (x86_64)
-- | --
Revision | 67da1aeb32cedd27634ca6634fb79cbd85d3f0ab-refs/branch-heads/4951@{#1126}
OS | macOS Version 12.3.1 (Build 21E258)
```

### Reproduced the issue in 1.38.x

Verify that the issuers were downloaded
<details>

```

```

</details>

Verify that unblinded tokens are downloaded
<details>

```

```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (US) - 
Brave Ads Resources (US) - 
Brave Ads Resources (en) - 


</details>

Viewed an ad and dismissed it

#### The memory used is KB




### Verified fixed in 1.39.x

Verify that the issuers were downloaded
<details>

```

```


</details>

Verify that unblinded tokens are downloaded
<details>

```

```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (US) - 
Brave Ads Resources (US) - 
Brave Ads Resources (en) - 

</details>

Viewed an ad and dismissed it

#### The memory used is KB, which is less than in 1.38.x

**Comment by MadhaviSeelam** [comment]
Verified `PASSED` using

```
Brave | 1.39.101 Chromium: 101.0.4951.64 (Official Build) beta (64-bit)
-- | --
Revision | d1daa9897e1bc1d507d6be8f2346e377e5505905-refs/branch-heads/4951@{#1208}
OS | Windows 11 Version 21H2 (Build 22000.613)

```

<details>
<summary>Upgrade case - PASSED</summary>

### Reproduced the issue in 1.38.x

Verify that the issuers were downloaded
<details>

```
[23196:9540:0512/110429.224:VERBOSE1:ad_server.cc(156)] Fetch catalog in 0 hours, 54 minutes, 46 seconds at 11:59:15.221[23196:9540:0512/110429.225:VERBOSE1:issuers.cc(80)] OnGetIssuers
[23196:9540:0512/110429.225:VERBOSE6:issuers.cc(82)] URL Response:
  URL: https://static.ads.brave.com/v1/issuers/
  Response Status Code: 200
```
</details>


Verify that unblinded tokens are downloaded
<details>

```

**Comment by atuchin-m** [comment]
@LaurenWags @LaurenWags @btlechowski Big thanks for verification!

Just want to clarify one thing: `Memory footprint` on Linux/Mac is counted in a different way that on Windows.
That's why it's ok that absolute numbers are not match across OSes, so please don't be confused by this.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 12` using the following build(s):

```
Brave | 1.38.119 Chromium: 101.0.4951.67 (Official Build) (64-bit)
--- | ---
Revision | 8888ee7a24e2c36661ddb9536c35b7d4852a3a98-refs/branch-heads/4951@{#1230}
OS | Android 12; Build/SP1A.210812.016
```

### Reproduced the issue in 1.38.x

Verify that the issuers were downloaded

<details>

```
2022-05-23 12:04:28.931 31131-31131/? V/chromium: [VERBOSE1:issuers.cc(80)] OnGetIssuers
2022-05-23 12:04:28.931 31131-31131/? V/chromium: [VERBOSE6:issuers.cc(82)] URL Response:
URL: https://ads-serve.brave.com/v1/issuers/
Response Status Code: 200
```
</details>

Verify that unblinded tokens are downloaded

<details>

```
2022-05-23 12:04:43.008 31131-31131/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(323)] Added 50 unblinded tokens, you now have 50 unblinded tokens
2022-05-23 12:04:43.009 31131-31131/? V/chromium: [VERBOSE1:account.cc(292)] Successfully refilled unblinded tokens
```
</details>

Verified ad related components are downloaded in `brave://components`

<details>

NTP Sponsored Images (CA) - Version: 1.0.149
Brave Ads Resources (CA) - Version: 1.0.49
Brave Ads Resources (en) - Version: 1.0.41

![Screenshot_20220523-124501_Brave](https://user-images.githubusercontent.com/2602313/169870536-02eafaad-b550-4100-bf40-97469e799176.jpg)
</details>

Viewed an ad and dismissed it

#### The memory used is `39.9 MiB`

![broken](https://user-images.githubusercontent.com/2602313/169868220-ef1b4070-ca58-445c-a71e-958fb98d237f.png)

### Verified fixed in 1.39.x via Clean Install

Verify that the issuers were downloaded
<details>

```
2022-05-23 12:35:06.549 12198-12198/? V/chromium: [VERBOSE1:issuers.cc(80)] OnGetIssuers
2022-05-23 12:35:06.549 12198-12198/? V/chromium: [VERBOSE6:issuers.cc(82)] URL Response:
URL: https://ads-serve.brave.com/v1/issuers/
Response Status Code: 200
```
</details>

Verify that unblinded tokens are downloaded
<details>

```
2022-05-23 13:08:52.085 23957-23957/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(322)] Added 50 unblinded tokens, you now have 50 unblinded tokens
2022-05-23 13:08:52.085 23957-23957/? V/chromium: [VERBOSE1:account.cc(291)] Successfully refilled unblinded tokens
```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (CA) - Version: 1.0.149
Brave Ads Resources (CA) - Version: 1.0.49
Brave Ads Resources (en) - Version: 1.0.41

![Screenshot_20220523-132254_Brave](https://user-images.githubusercontent.com/2602313/169874223-022b853c-1514-4969-8b2a-909e0cd09671.jpg)

</details>

Viewed an ad and dimissed it

#### The memory used is `22.3 MiB`, which is less than in 1.38.x

![image](https://user-images.githubusercontent.com/2602313/169873747-d06a6528-ba82-44b4-8c73-ad615af66709.png)

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung Galaxy Tablet A` running `Android 11` using the following build(s):

```
Brave | 1.39.111 Chromium: 102.0.5005.61 (Official Build) (64-bit)
--- | ---
Revision | 0e59bcc00cc4985ce39ad31c150065f159d95ad3-refs/branch-heads/5005@{#819}
OS | Android 11; Build/RP1A.200720.012
```

### Reproduced the issue in 1.38.x

Verify that the issuers were downloaded

<details>

```
2022-05-23 14:06:17.345 9005-9005/? V/chromium: [VERBOSE1:issuers.cc(80)] OnGetIssuers
2022-05-23 14:06:17.346 9005-9005/? V/chromium: [VERBOSE6:issuers.cc(82)] URL Response:
URL: https://ads-serve.brave.com/v1/issuers/
Response Status Code: 200
```
</details>

Verify that unblinded tokens are downloaded

<details>

```
2022-05-23 14:29:26.897 18263-18263/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(323)] Added 50 unblinded tokens, you now have 50 unblinded tokens
2022-05-23 14:29:26.897 18263-18263/? V/chromium: [VERBOSE1:account.cc(292)] Successfully refilled unblinded tokens
```
</details>

Verified ad related components are downloaded in `brave://components`

<details>

NTP Sponsored Images (CA) - Version: 1.0.149
Brave Ads Resources (CA) - Version: 1.0.49
Brave Ads Resources (en) - Version: 1.0.41

![Screenshot_20220523-141216_Brave](https://user-images.githubusercontent.com/2602313/169881386-1bd2c076-58fc-4beb-9a3f-2d818c13c2b2.jpg)
</details>

Viewed an ad and dismissed it

#### The memory used is `33.1 MiB`

![image](https://user-images.githubusercontent.com/2602313/169880984-d1be296a-bd86-4930-a460-13393125d06a.png)

### Verified fixed in 1.39.x via Clean Install

Verify that the issuers were downloaded
<details>

```
2022-05-23 14:38:39.641 24767-24767/? V/chromium: [VERBOSE1:issuers.cc(80)] OnGetIssuers
2022-05-23 14:38:39.641 24767-24767/? V/chromium: [VERBOSE6:issuers.cc(82)] URL Response:
URL: https://static.ads.brave.com/v1/issuers/
Response Status Code: 200
```
</details>

Verify that unblinded tokens are downloaded
<details>

```
2022-05-23 14:38:45.407 24767-24767/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(322)] Added 50 unblinded tokens, you now have 50 unblinded tokens
2022-05-23 14:38:45.407 24767-24767/? V/chromium: [VERBOSE1:account.cc(291)] Successfully refilled unblinded tokens
```

</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (CA) - Version: 1.0.149
Brave Ads Resources (CA) - Version: 1.0.49
Brave Ads Resources (en) - Version: 1.0.41

![Screenshot_20220523-144019_Brave](https://user-images.githubusercontent.com/2602313/169885742-164d9b42-d182-431f-bb15-7e63e6eb7d49.jpg)

</details>

Viewed an ad and dimissed it

#### The memory used is `6.8 MiB`, which is less than in 1.38.x

![image](https://user-images.githubusercontent.com/2602313/169885521-817e6f4e-2145-49b4-9e1b-c7462aec4069.png)

### Verified fixed in 1.39.x via Upgrade

Verify that the issuers were downloaded using 1.38.x
<details>

```
2022-05-23 14:06:17.345 9005-9005/? V/chromium: [VERBOSE1:issuers.cc(80)] OnGetIssuers
2022-05-23 14:06:17.346 9005-9005/? V/chromium: [VERBOSE6:issuers.cc(82)] URL Response:
URL: https://ads-serve.brave.com/v1/issuers/
Response Status Code: 200
```

</details>

Verify that unblinded tokens are downloaded using 1.38.x
<details>

```
2022-05-23 14:29:26.897 18263-18263/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(323)] Added 50 unblinded tokens, you now have 50 unblinded tokens
2022-05-23 14:29:26.897 18263-18263/? V/chromium: [VERBOSE1:account.cc(292)] Successfully refilled unblinded tokens
```
</details>

Verified ad related components are downloaded in `brave://components`
<details>

NTP Sponsored Images (CA) - Version: 1.0.149
Brave Ads Resources (CA) - Version: 1.0.49
Brave Ads Resources (en) - Version: 1.0.41

![Screenshot_20220523-141216_Brave](https://user-images.githubusercontent.com/2602313/169881386-1bd2c076-58fc-4beb-9a3f-2d818c13c2b2.jpg)

</details>

Viewed an ad and dismissed using 1.38.x

#### The memory used is `40.3 MiB`, which is less than in 1.38.x

![image](https://user-images.githubusercontent.com/2602313/169878011-136c572e-f3c7-4cc1-80ee-524ab04917e1.png)

Upgraded to `1.39.x` and viewed/dismissed and ad as per the following:

#### The memory used is `12.9 MiB`, which is less than in 1.38.x

![image](https://user-images.githubusercontent.com/2602313/169883063-7540881f-da4e-4eb3-9ee5-7c60834da976.png)
